src/Services/SrvNinjaHelper.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Core;
+using Core.Services;
+using Lykke.Service.Assets.Client.Custom;
+using Services.NinjaContracts;
+
+namespace Services
+{
+    /// <summary>
+    /// The class responsible for communication with Bitcoin Ninja server.
+    /// </summary>
+    public class SrvNinjaHelper : ISrvNinjaHelper
+    {
+        private readonly ICachedAssetsService _assetsService;
+        private readonly string _url;
+
+        #region Initialization
+
+        public SrvNinjaHelper(ICachedAssetsService assetsService, string url)
+        {
+            _assetsService = assetsService ?? throw new ArgumentNullException(nameof(assetsService));
+
+            _url = 
+                !string.IsNullOrWhiteSpace(url)
+                ? url
+                : throw new ArgumentNullException(nameof(url));
+        }
+
+        #endregion
+
+        #region Public
+
+        /// <inheritdoc cref="ISrvNinjaHelper"/>
+        public async Task<double> GetBalance(string address, string assetName)
+        {
+            if (string.IsNullOrWhiteSpace(address))
+                throw new ArgumentNullException(nameof(address));
+            if (string.IsNullOrWhiteSpace(assetName))
+                throw new ArgumentNullException(nameof(assetName));
+
+            var assets = await _assetsService.GetAllAssetsAsync();
+            if (!assets?.Any() ?? false)
+                throw new IOException("Could not obtain the list of available assets.");
```

Comment:
This is not so imprtant here, but It's not an IOException. InvalidOperationException will be ok